pymc_marketing/clv/models/pareto_nbd.py

```diff
@@ -0,0 +1,589 @@
+        )
+        values = np.vstack((t_x.values, x.values)).T
+        loglike = pm.logp(pareto_dist, values).eval()
+        loglike = xarray.DataArray(data=loglike, dims=("chain", "draw", "customer_id"))
+
+        term1 = gammaln(r + x) - gammaln(r)
+        term2 = r * log(alpha / (alpha + T))
+        term3 = -x * log(alpha + T)
+        term4 = s * log(beta / (beta + T + future_t))
+
+        prob_alive = exp(term1 + term2 + term3 + term4 - loglike)
+
+        return prob_alive.transpose(
+            "chain", "draw", "customer_id", missing_dims="ignore"
+        )
+
+    # TODO: Edit docstrings
+    def purchase_probability(
+        self,
+        n: Union[int, np.ndarray, pd.Series, TensorVariable],
+        future_t: Union[float, np.ndarray, pd.Series, TensorVariable],
+        customer_id: Union[np.ndarray, pd.Series] = None,
+        frequency: Union[np.ndarray, pd.Series, TensorVariable] = None,
+        recency: Union[np.ndarray, pd.Series, TensorVariable] = None,
+        T: Union[np.ndarray, pd.Series, TensorVariable] = None,
+    ) -> xarray.DataArray:
+        """
+        Return conditional probability of n purchases up to time t.
+
+        Calculate the probability of n purchases up to time t for an individual
+        with history frequency, recency and T (age).
+
+        The main equation being implemented is (16) from:
+        http://www.brucehardie.com/notes/028/pareto_nbd_conditional_pmf.pdf
+
+        Parameters
+        ----------
+        n: int
+            number of purchases.
+        t: a scalar
+            time up to which probability should be calculated.
+        frequency: float
+            historical frequency of customer.
+        recency: float
+            historical recency of customer.
+        T: float
+            age of the customer.
+
+        Returns
+        -------
+        array_like
+        """
+
+        if future_t <= 0:
+            return 0
+
+        if customer_id is None:
+            customer_id = self._customer_id
+        if frequency is None:
+            frequency = self._frequency
+        if recency is None:
+            recency = self._recency
+        if T is None:
+            T = self._T
+
+        x, t_x, T = to_xarray(customer_id, frequency, recency, T)
+
+        r, alpha, s, beta = self._unload_params()
+
+        # Get likelihood expression
+        # Add one dummy dimension to the right of the scalar parameters, so they broadcast with the `T` vector
+        pareto_dist = ParetoNBD.dist(
+            r=r.values[..., None],
+            alpha=alpha.values[..., None],
+            s=s.values[..., None],
+            beta=beta.values[..., None],
+            T=T.values,
+        )
+        values = np.vstack((t_x.values, x.values)).T
+        loglike = pm.logp(pareto_dist, values).eval()
+        loglike = xarray.DataArray(data=loglike, dims=("chain", "draw", "customer_id"))
+
+        # TODO: Are these mean comparisons kosher?
+        if alpha.mean() < beta.mean():
```

Comment:
I just read through the math below and it seems to align with (16). I also see why you have the if/else block due to how B2 and B3i are defined in equations (18) and (19).

I'm just not so sure about the `alpha.mean() < beta.mean()` comparison... In cases where they have close values, would we want `min_of_alpha_beta` and `max_of_alpha_beta` to have values within their vectors to vary depending on the draw values of `alpha` and `beta`? Perhaps that's what you meant by the comment above...

Reply:
As for the shape-related error that you're getting in the CI, I'll need to have a closer look...